### PR TITLE
Replace while-sleep with Event

### PR DESCRIPTION
**Pull request by Olegt0rr**
# Description

Instead of sleeping in a while loop, and waiting for a condition to become true, it's preferable to await on an Event object such as: `anyio.Event`

This way you don't bother event loop every `0.1` seconds (used as default) and handle condition change without waiting additional sleep time. 

## Type of change

Please delete options that are not relevant.

- [ ] Documentation (typos, code examples, or any documentation updates)
- [ ] Bug fix (a non-breaking change that resolves an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would disrupt existing functionality)
- [ ] This change requires a documentation update

## Checklist

- [x] My code adheres to the style guidelines of this project (`scripts/lint.sh` shows no errors)
- [x] I have conducted a self-review of my own code
- [ ] I have made the necessary changes to the documentation
- [ ] My changes do not generate any new warnings
- [ ] I have added tests to validate the effectiveness of my fix or the functionality of my new feature
- [ ] Both new and existing unit tests pass successfully on my local environment by running `scripts/test-cov.sh`
- [ ] I have ensured that static analysis tests are passing by running `scripts/static-analysis.sh`
- [ ] I have included code examples to illustrate the modifications

### CODE REVIEW

File: faststream/app.py [hunk 1]
```diff
@@ -236,3 +238,17 @@ async def _shutdown(self, log_level: int = logging.INFO) -> None:
...
+        return self._should_exit.is_set()
+
+    @should_exit.setter
+    def should_exit(self, value: Literal[True]) -> None:
+        """Set should exit flag state."""
+        if value is not True:
+            msg = "'should_exit' can receive only 'True' value"
+            raise ValueError(msg)
+
+        self._should_exit.set()
```
Comment (Olegt0rr) [inline faststream/app.py:254]: I'm not sure if you want to save `should_exit` public boolean or not

File: faststream/app.py [hunk 2]
```diff
@@ -154,7 +153,13 @@ async def run(
         self,
         log_level: int = logging.INFO,
         run_extra_options: Optional[Dict[str, "SettingField"]] = None,
-        sleep_time: float = 0.1,
+        sleep_time: Annotated[
+            float,
+            deprecated(
+                "Deprecated in **FastStream 0.5.0**. "
```
Comment (Olegt0rr) [inline faststream/app.py:159]: Should I set patch-version here? What version should I set?

**Comment by Lancetnik** [comment]
Sorry, we should take a little pause with this PR. Our first implementation worked this way, but we switched to sleeping due `nats-py` bug. We should check it again

File: faststream/app.py (hunk 1 above)
Comment (Lancetnik) [reply faststream/app.py:254]: No, just remove it

**Comment by Lancetnik** [comment]
@Olegt0rr I think, we can merge it and come back to the problem if there is a feedback from NATS users
Can u please resolve conflict and remove property to let me merge PR?

File: faststream/app.py (hunk 2 above)
Comment (Lancetnik) [reply faststream/app.py:159]: 0.5.24

File: faststream/app.py (hunk 1 above)
Comment (Lancetnik) [reply faststream/app.py:254]: Just remove it :)

**Review by Lancetnik — APPROVED** [review]

**Pull request merged**